docs/operator-guide/upgrade/upgrade-edp-3.10.1.md

```diff
@@ -0,0 +1,251 @@
+    kubectl apply -f https://raw.githubusercontent.com/epam/edp-codebase-operator/v2.24.0/deploy-templates/crds/v2.edp.epam.com_jiraservers.yaml
+    kubectl apply -f https://raw.githubusercontent.com/epam/edp-cd-pipeline-operator/v2.21.0/deploy-templates/crds/v2.edp.epam.com_cdpipelines.yaml
+    kubectl apply -f https://raw.githubusercontent.com/epam/edp-cd-pipeline-operator/v2.21.0/deploy-templates/crds/v2.edp.epam.com_stages.yaml
+    ```
+
+2. (Optional) Update the CRDs for Gerrit if **gerrit-operator** is being used. Run the following command to apply the necessary CRDs to the cluster:
+
+    ```bash
+    kubectl apply -f https://raw.githubusercontent.com/epam/edp-gerrit-operator/v2.22.0/deploy-templates/crds/v2.edp.epam.com_gerritgroupmembers.yaml
+    kubectl apply -f https://raw.githubusercontent.com/epam/edp-gerrit-operator/v2.22.0/deploy-templates/crds/v2.edp.epam.com_gerritgroups.yaml
+    kubectl apply -f https://raw.githubusercontent.com/epam/edp-gerrit-operator/v2.22.0/deploy-templates/crds/v2.edp.epam.com_gerritmergerequests.yaml
+    kubectl apply -f https://raw.githubusercontent.com/epam/edp-gerrit-operator/v2.22.0/deploy-templates/crds/v2.edp.epam.com_gerritprojectaccesses.yaml
+    kubectl apply -f https://raw.githubusercontent.com/epam/edp-gerrit-operator/v2.22.0/deploy-templates/crds/v2.edp.epam.com_gerritprojects.yaml
+    kubectl apply -f https://raw.githubusercontent.com/epam/edp-gerrit-operator/v2.22.0/deploy-templates/crds/v2.edp.epam.com_gerritreplicationconfigs.yaml
+    kubectl apply -f https://raw.githubusercontent.com/epam/edp-gerrit-operator/v2.22.0/deploy-templates/crds/v2.edp.epam.com_gerrits.yaml
+    ```
+
+3. (Optional) If the **External Secret Operator** is used for managing secrets, specify two new fields in the `values.yaml` file: `manageGitProviderSecretsName` and `manageCodemieSecretsName`. These fields should be defined as follows:
+
+    ```yaml title="values.yaml"
+    externalSecrets:
+      enabled: true
+      ...
+      manageCodemieSecretsName: /edp/codemie-secrets
+      manageGitProviderSecretsName: /edp/git-provider-secrets
+    ```
+
+    These objects must be pre-created in AWS ParameterStore. Below are examples of the object structures.
+
+    For `manageGitProviderSecretsName` AWS Parameter Store object:
+
+        ```json
+        {
+          "github-config": {
+            "id_rsa": "LS0tLS1...",
+            "token": "ghp_fdO1joO...",
+            "secretString": "kfr1ndk4..."
+          },
+          "gitlab-config": {
+            "id_rsa": "LS0tLS1...",
+            "token": "RW1UG2EAdEjPEC1NSoVm3A",
+            "secretString": "kfr1ndk4..."
+          },
+          "bitbucket-config": {
+            "id_rsa": "LS0tLS1...",
+            "token": "YXV0b19lcG1kLWVkc...",
+            "secretString": "kfr1ndk4..."
+          }
+        }
+        ```
+    For `manageCodemieSecretsName` AWS Parameter Store object:
```

Comment:
codemi new feature for this release or user can use in previous release? in this guide we need to higlight only necessary steps. so in CodeMie available only in this release we do not need to highlight this in gudie